### PR TITLE
Log errors for VS Code debugger

### PR DESCRIPTION
**Pull request by laurakwhit**
## What was changed

Get data from responses in order to log error messages.

## Why?

https://github.com/temporalio/vscode-debugger-plugin/pull/22#discussion_r1019718787

## Checklist

1. Closes: n/a 

2. How was this tested: n/a

3. Any docs updates needed? n/a

### CODE REVIEW

File: packages/worker/src/debug-replayer/client.ts [hunk 1]
```diff
@@ -41,10 +41,19 @@ export class Client {
...
+      request.once('response', (res) => {
+        let body = [];
+        res.on('data', function (chunk) {
+          body.push(chunk);
+        });
+        res.on('end', function () {
+          res.body = JSON.parse(body.toString());
+          resolve(res);
+        });
+      });
```
Comment (bergundy) [inline packages/worker/src/debug-replayer/client.ts:53]: You can use the response as an async iterator if you need the body.
In some cases we don't want to consume the body as JSON, I would only consume it on error.

Comment (laurakwhit) [reply packages/worker/src/debug-replayer/client.ts]: That makes total sense. Updated ✅

File: packages/worker/src/debug-replayer/client.ts [hunk 2]
```diff
@@ -44,7 +44,9 @@ export class Client {
       request.once('response', resolve);
     });
     if (response.statusCode !== 200) {
-      throw new Error(`Bad response code from VS Code: ${response.statusCode}`);
+      const responseBody = await Client.readAll(response);
+      const message = JSON.parse(responseBody.toString())?.error ?? response.statusMessage;
```
Comment (bergundy) [inline packages/worker/src/debug-replayer/client.ts:48]: Oh I didn't realize we already had this `readAll` helper..

One last thing here.. Wrap the `JSON.parse` statement in `try .. catch`.
It's not safe to assume that all response bodies would contain valid JSON.

Comment (laurakwhit) [reply packages/worker/src/debug-replayer/client.ts:48]: Ah, good point. I think we can just ignore the error in that case? e.g.

```js
      let message = response.statusMessage;
      try {
        message = JSON.parse(responseBody.toString())?.error ?? message;
      } catch {
        // ignore
      }
```

File: packages/worker/src/debug-replayer/client.ts [hunk 3]
```diff
@@ -44,7 +44,14 @@ export class Client {
       request.once('response', resolve);
     });
     if (response.statusCode !== 200) {
-      throw new Error(`Bad response code from VS Code: ${response.statusCode}`);
+      const responseBody = await Client.readAll(response);
```
Comment (bergundy) [inline packages/worker/src/debug-replayer/client.ts:47]: I hate to do request another change but can you please move reading the body to the try block?
Not super critical, I'm approving the PR and leaving it up to you.

**Review by bergundy — APPROVED** [review]

File: packages/worker/src/debug-replayer/client.ts (hunk 3 above)
Comment (laurakwhit) [reply packages/worker/src/debug-replayer/client.ts]: Done, thank you ✅

**Comment by bergundy** [comment]
Thanks @laurakwhit!

**Pull request merged**